### PR TITLE
[1.11.x] PLANNER-2474 optaplanner CI (#215)

### PR DESCRIPTION
**Pull request by radtriste**
# JIRA
https://issues.redhat.com/browse/PLANNER-2474

# Cherry-pick
https://github.com/kiegroup/kogito-pipelines/pull/215

# Related PRs:
* https://github.com/kiegroup/kogito-pipelines/pull/246
* https://github.com/kiegroup/optaplanner/pull/1529
* https://github.com/kiegroup/optaplanner-quickstarts/pull/165
* https://github.com/kiegroup/optaweb-employee-rostering/pull/681
* https://github.com/kiegroup/optaweb-vehicle-routing/pull/551

### CODE REVIEW

File: .ci/project-dependencies.yaml [hunk 1]
```diff
@@ -13,14 +25,96 @@ dependencies:
...
+    mapping:
+      dependencies:
+        default:
+          - source: development
+            target: main
+        kiegroup/kogito-runtimes:
+          - source: (8)\.(.*)\.(.*)
+            targetExpression: "process.env.GITHUB_BASE_REF.replace(/(\\d*)\\.(.*)\\.(.*)/g, (m, n1, n2, n3) => `${+n1-7}.${n2}.${n3}`)"
+          - source: development
+            target: main
```
Comment (radtriste) [inline .ci/project-dependencies.yaml:114]: This part will need to be backported to `main` branch as long as https://github.com/kiegroup/github-action-build-chain/issues/168 is not solved

**Review by rsynek — COMMENTED** [review]
I just quickly scrolled through this PR and LGTM. @Ginxo can hopefully review this in depth.

File: .ci/project-dependencies.yaml (hunk 1 above)
Comment (radtriste) [reply .ci/project-dependencies.yaml:114]: https://github.com/kiegroup/kogito-pipelines/pull/268

File: .ci/jenkins/Jenkinsfile.buildchain [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            steps {
+                script {
+                    if (isSonarCloudAnalysis()) {
+                        env.BUILD_MVN_OPTS_CURRENT = "${env.BUILD_MVN_OPTS_CURRENT ?: ''} -Prun-code-coverage"
+                    }
+                    def buildChainActionInfo = getBuildChainActionInfo()
+
+                    configFileProvider([configFile(fileId: getSettingsXmlId(), variable: 'MAVEN_SETTINGS_FILE')]) {
+                        withCredentials([string(credentialsId: 'kie-ci1-token', variable: 'GITHUB_TOKEN')]) {
+                            sh "build-chain-action -token=${GITHUB_TOKEN} -df='https://raw.githubusercontent.com/\${GROUP}/kogito-pipelines/main/.ci/${buildChainActionInfo.file}' -folder='bc' build ${buildChainActionInfo.action} -url=${env.ghprbPullLink} ${buildChainActionInfo.arguments} --skipParallelCheckout -cct '(^mvn .*)||\$1 -s ${MAVEN_SETTINGS_FILE} -Dmaven.wagon.http.ssl.insecure=true'"
```
Comment (Ginxo) [inline .ci/jenkins/Jenkinsfile.buildchain]: is `https://raw.githubusercontent.com/\${GROUP}/kogito-pipelines/main` the right branch? shouldn't it be `1.11.x`?

**Review by Ginxo — CHANGES_REQUESTED** [review]
please check

File: .ci/jenkins/Jenkinsfile.buildchain (hunk 2 above)
Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.buildchain]: true but note that the prepare release branch will not solve that when creating a new branch ...

Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.buildchain]: I am just thinking. it could be a value we pass as env variable. So we don't need to hardcode. wdyt ?

Comment (Ginxo) [reply .ci/jenkins/Jenkinsfile.buildchain]: @radtriste I agree on using env var there, we should apply the same in main branch.

Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.buildchain]: I will harcode that here and do a PR on `main` for the next release branches

Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.buildchain]: https://github.com/kiegroup/kogito-pipelines/pull/270

**Pull request merged**